api/tests/Feature/CandidateAssessmentStatusTest.php

```diff
@@ -0,0 +1,344 @@
+use Tests\UsesProtectedGraphqlEndpoint;
+
+class CandidateAssessmentStatusTest extends TestCase
+{
+    use MakesGraphQLRequests;
+    use RefreshDatabase;
+    use RefreshesSchemaCache;
+    use UsesProtectedGraphqlEndpoint;
+
+    protected $pool;
+
+    protected $adminUser;
+
+    protected $candidate;
+
+    protected $queryVars;
+
+    protected $query = /** GraphQL */
+        '
+        query poolCandidate($id: UUID!) {
+            poolCandidate(id: $id) {
+                assessmentStatus {
+                    currentStep
+                    finalDecision
+                    stepDecisions {
+                        step
+                        decision
+                    }
+                }
+            }
+        }
+    ';
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->seed([
+            SkillFamilySeeder::class,
+            SkillSeeder::class,
+            RolePermissionSeeder::class,
+        ]);
+
+        $team = Team::factory()->create([
+            'name' => 'assessment-status',
+        ]);
+
+        $this->pool = Pool::factory()
+            ->withPoolSkills(1, 0)
+            ->withAssessments()
+            ->published()
+            ->create([
+                'team_id' => $team->id,
+            ]);
+
+        $this->adminUser = User::factory()
+            ->asApplicant()
+            ->asPoolOperator('assessment-status')
+            ->asRequestResponder()
+            ->asAdmin()
+            ->create();
+
+        $this->candidate = PoolCandidate::factory()->create([
+            'pool_id' => $this->pool->id,
+            'submitted_at' => config('constants.past_date'),
+            'expiry_date' => config('constants.far_future_date'),
+        ]);
+
+        $this->queryVars = [
+            'id' => $this->candidate->id,
+        ];
+    }
+
+    public function testDefaultAssessmentStatus(): void
+    {
+        $this->actingAs($this->adminUser, 'api')
+            ->graphQL($this->query, $this->queryVars)
+            ->assertJson([
+                'data' => [
+                    'poolCandidate' => [
+                        'assessmentStatus' => [
+                            'stepDecisions' => [],
+                            'finalDecision' => AssessmentFinalDecision::TO_ASSESS->name,
+                            'currentStep' => 1,
+                        ],
+                    ],
+                ],
+            ]);
+    }
+
+    public function testSuccessPassesFirstStep(): void
+    {
+        $stepOne = $this->pool->assessmentSteps->where('sort_order', 1)->first();
+
+        AssessmentResult::factory()
+            ->withResultType(AssessmentResultType::EDUCATION)
+            ->create([
+                'assessment_step_id' => $stepOne->id,
+                'pool_candidate_id' => $this->candidate->id,
+                'assessment_decision' => AssessmentDecision::SUCCESSFUL->name,
+            ]);
+
+        $this->actingAs($this->adminUser, 'api')
```

Comment:
Ah, yeah good catch. The essential skill assessment logic was not good. Fixed in [adjust essential skill assessment logic](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10642/commits/526d35308dfb0126873f31285205e2a249ef4c24)

Reply:
Sorry, I'm still confused about this. The first step should be an APPLICATION_ASSESSMENT step, which always requires an education assessment as well as one for every required technical skill. Why is the skill assessment not required here?

Reply:
Huh, yeah. I was able to get this to fail but I think things have changed since then. I'll nee to revisit it :confounded:

Reply:
I just had a thought. When a Pool is created, the `created` event automatically adds an APPLICATION_SCREENING step. But I don't see the code which adds all technical skills to the APPLICATION_SCREENING step whenever they're added to the Pool. Are we missing a listener which does that, and ending up with an APPLICATION_SCREENING step which is missing some relevant related skills?

Reply:
That is done by this function

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/455d0eb2-9e10-4dc7-a197-4d4649d84f7d)

Called by

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/717c551d-cc90-4a90-8671-0db99431bc64)

I believe

Reply:
Okay, I found it.. even though the result is of type education, it still assesses the skill so it will pass. Since the UI does not allow for assessing skills on education I have added that as a qualifier for the essential skill assessments. 

[fix education with skill assessments affecting skill completion](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10642/commits/38abe59a092c9adc50821fa111dac87e03ad0618)

Now, if you comment out the skill assessment in this test, it will fail as expected.